server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/service/CassandraMigrationService.java

```diff
@@ -67,32 +69,36 @@ public synchronized void upgradeToVersion(int newVersion) {
 
         IntStream.range(currentVersion, newVersion)
             .boxed()
-            .forEach(this::doMigration);
+            .forEach(Throwing.consumer(this::doMigration));
     }
 
     public void upgradeToLastVersion() {
         upgradeToVersion(latestVersion);
     }
 
-    private void doMigration(Integer version) {
+    private void doMigration(Integer version) throws MailboxException {
+        int newVersion = version + 1;
         if (allMigrationClazz.containsKey(version)) {
-            LOG.info("Migrating to version {} ", version + 1);
-            Migration.Result migrationResult = allMigrationClazz.get(version).run();
-            if (migrationResult == Migration.Result.COMPLETED) {
-                schemaVersionDAO.updateVersion(version + 1);
-                LOG.info("Migrating to version {} done", version + 1);
-            } else {
-                String message = String.format("Migrating to version %d partially done. " +
-                    "Please check logs for cause of failure and re-run this migration.",
-                    version + 1);
-                LOG.warn(message);
-                throw new MigrationException(message);
-            }
+            LOG.info("Migrating to version {} ", newVersion);
+            allMigrationClazz.get(version).run()
+                .ifCompleted(() -> schemaVersionDAO.updateVersion(newVersion))
```

Comment:
why don't we use classical "onComplete" and "onFailture" instead ? onFailure could give you the errors, too

Reply:
and we could have a multiline lambda instead of calling twice the same method